### ISSUE TITLE
CUDA error: all CUDA-capable devices are busy or unavailable

### ISSUE THREAD

**Issue by SchordingersDogCat** [pytorch/pytorch#58824]
Hi,

I am using Pytorch 1.8.1+cu102 on a Ubuntu system.

I have already setup the CUDA and GPU driver. Everything goes very well because I can directly access the GPU by writing some CUDA code.

However, when I try to use pytorch, the return of  “torch.cuda.is_available” is True. But when I try to access the "torch.device('cuda')", I got the following error:

all CUDA-capable devices are busy or unavailable

**Comment by zou3519** [comment]
Could you run and paste the output of https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py?

**Comment by SchordingersDogCat** [comment]
Hi,

Finally I walk around the issue by updating the CUDA driver. The previous CUDA version is 11.0. Then I upgraded it to 11.3. The problem is solved 

I did the following works to debug the problem:

1. I checked whether the GPU is set as the shared mode. I find the GUI is also using the GPU. So the mode is set correctly.
2. I wrote a simple CUDA program and it works quite well. So the CUDA works very well.
3. Then I realize the problem lies in the Pytorch. I tried to upgrade Pytorch to the right version as CUDA. But the problem still exists. I used the version Pytorch 1.8.1+cu110.
4. Finally, I upgraded my CUDA driver. The problem is solved.

The debugging process is quite time consuming. But I did learn quite a lot on how the CUDA works.
